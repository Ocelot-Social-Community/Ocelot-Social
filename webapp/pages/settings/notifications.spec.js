import Vuex from 'vuex'
import { mount } from '@vue/test-utils'
import Notifications from './notifications.vue'

const localVue = global.localVue

describe('notifications.vue', () => {
  let wrapper
  let mocks
  let store

  beforeEach(() => {
    mocks = {
      $t: jest.fn(),
      $apollo: {
        mutate: jest.fn(),
      },
      $toast: {
        success: jest.fn(),
        error: jest.fn(),
      },
    }
    store = new Vuex.Store({
      getters: {
        'auth/user': () => {
          return {
            id: 'u343',
            name: 'MyAccount',
            sendNotificationEmails: true,
          }
        },
      },
    })
  })

  describe('mount', () => {
    const Wrapper = () => {
      return mount(Notifications, {
        store,
        mocks,
        localVue,
      })
    }

    beforeEach(() => {
      wrapper = Wrapper()
    })

    it('renders', () => {
      expect(wrapper.classes('base-card')).toBe(true)
    })

    it('clicking on submit changes notifyByEmail to false', async () => {
      await wrapper.find('#send-email').setChecked(false)
      await wrapper.find('.base-button').trigger('click')
      expect(wrapper.vm.notifyByEmail).toBe(false)
    })

    it('clicking on submit with a server error shows a toast and notifyByEmail is still true', async () => {
      mocks.$apollo.mutate = jest.fn().mockRejectedValue({ message: 'Ouch!' })
      await wrapper.find('#send-email').setChecked(false)
      await wrapper.find('.base-button').trigger('click')
      expect(mocks.$toast.error).toHaveBeenCalledWith('Ouch!')
      expect(wrapper.vm.notifyByEmail).toBe(true)
    })
  })
})
