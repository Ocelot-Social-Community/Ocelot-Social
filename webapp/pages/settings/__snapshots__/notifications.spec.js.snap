// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`notifications.vue mount renders 1`] = `
<article class="base-card">
  <h2 class="title">settings.notifications.name</h2>
  <div class="ds-space" style="margin-top: 16px; margin-bottom: 16px;">
    <div class="ds-space" style="margin-bottom: 8px;">
      <h4>settings.notifications.post</h4>
    </div>
    <div class="notifcation-settings-section">
      <div class="ds-space" style="margin-bottom: 8px;"><input id="commentOnObservedPost" type="checkbox"> <label for="commentOnObservedPost" class="label">
          settings.notifications.commentOnObservedPost
        </label></div>
      <div class="ds-space" style="margin-bottom: 8px;"><input id="mention" type="checkbox"> <label for="mention" class="label">
          settings.notifications.mention
        </label></div>
    </div>
  </div>
  <div class="ds-space" style="margin-top: 16px; margin-bottom: 16px;">
    <div class="ds-space" style="margin-bottom: 8px;">
      <h4>settings.notifications.group</h4>
    </div>
    <div class="notifcation-settings-section">
      <div class="ds-space" style="margin-bottom: 8px;"><input id="groupMemberJoined" type="checkbox"> <label for="groupMemberJoined" class="label">
          settings.notifications.groupMemberJoined
        </label></div>
      <div class="ds-space" style="margin-bottom: 8px;"><input id="groupMemberLeft" type="checkbox"> <label for="groupMemberLeft" class="label">
          settings.notifications.groupMemberLeft
        </label></div>
      <div class="ds-space" style="margin-bottom: 8px;"><input id="groupMemberRemoved" type="checkbox"> <label for="groupMemberRemoved" class="label">
          settings.notifications.groupMemberRemoved
        </label></div>
      <div class="ds-space" style="margin-bottom: 8px;"><input id="groupMemberRoleChanged" type="checkbox"> <label for="groupMemberRoleChanged" class="label">
          settings.notifications.groupMemberRoleChanged
        </label></div>
    </div>
  </div>
  <div class="ds-space" style="margin-top: 24px; margin-bottom: 8px;"><button type="button" class="base-button">
      <!---->
      <!---->
      settings.notifications.checkAll
    </button> <button type="button" class="base-button">
      <!---->
      <!---->
      settings.notifications.uncheckAll
    </button> <button disabled="disabled" type="button" class="save-button base-button --filled">
      <!---->
      <!---->
      actions.save
    </button></div>
  <!---->
</article>
`;
